### ISSUE TITLE
setting up config

### ISSUE THREAD

**Issue by ElianoMarques** [rstudio/sparklyr#87]
Hi, 

I'm trying to change the config of the spark environment, similar to the way we do with SparkR package. 
```r
library(SparkR)
spark_env = list('spark.executor.memory' = '4g', 
                 'spark.executor.instances' = '4', 
                 'spark.executor.cores' = '4',
                 'spark.driver.memory' = '4g') 

sc <- sparkR.init(master = local[*], appName = "SparkR", sparkEnvir = spark_env, 
                  sparkJars=c("/home/sparkjars/spark-csv_2.10-1.3.0.jar", 
                              "/home/sparkjars/commons-csv-1.2.jar"))
```

In this package I did the following:

Inside the package folder there is a folder called config/config-template.yml and I changed that to:

```
default:
  # local-only configuration
  sparklyr.cores.local: !expr parallel::detectCores()
  sparklyr.shell.driver-memory: 4GB
  sparklyr.shell.executor-memory: 4GB
  sparklyr.sql.shuffle.partitions.local: !expr parallel::detectCores()
```

then I run:

```r
sc = spark_connect(master= "local[*]" )
#also tried sc = spark_connect(master= "local[*]" , config = list('sparklyr.shell.executor-memory'='4GB'))
```

I get the following error: 

```r
Error in instances[[scon$sconRef]] <- NULL : 
  attempt to select less than one element in OneIndex
```

Would be great if you could clarify how to define the config directly in r-studio and also comment on the error. 

Thank you

**Comment by jjallaire** [comment]
You use a config.yml file, please see these docs:

http://spark.rstudio.com/deployment.html#configuration

I'm not sure about the error but it may have already been fixed on master. @
javierluraschi
<https://github.com/rstudio/sparklyr/commits/master?author=javierluraschi> ?

**Comment by ElianoMarques** [comment]
If we run the below or setup the config 

```r
sc = spark_connect(master= "local[*]" , config = list('sparklyr.shell.driver-memory'='8g'))
```
It works. executor-memory is not part of the R settings for Spark. 

Thanks for quick reply.

**Comment by javierluraschi** [comment]
@ElianoMarques your original `config.yml` is perfectly valid, this uncovered a minor regression on the last two days. Providing a fix...

**Comment by javierluraschi** [comment]
@ElianoMarques Fixed in `0.2.20`.

In `0.2.20` you will have to pass the parameter as:

```
sc <- spark_connect(master = "local[*]", config = list(
   "sparklyr.shell.driver-memory" = "4G")
)
```

Or alternatively, if you want to read your default config file and only modify a subset of the parameters you can use:

```
config <- spark_config()
config[["sparklyr.shell.driver-memory"]] <- "4G"
sc <- spark_connect(master = "local[*]", config = config)
```

**Comment by ElianoMarques** [comment]
Thanks
